tensorflow_lite_support/c/task/common_error_utils.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/common_error_utils.h"
+
+#include <string>
+
+#include "external/com_google_absl/absl/strings/cord.h"
+#include "tensorflow_lite_support/cc/common.h"
+
+namespace tflite {
+namespace task {
+
+namespace {
+using ::tflite::support::kTfLiteSupportPayload;
+}  // namespace
+
+void CreateTfLiteErrorWithStatus(const absl::Status& status,
+                                 TfLiteError** error) {
+  if (status.ok() or error == nullptr) return;
+
+  // Payload of absl::Status created by the tflite task library stores an
+  // appropriate value of the enum TfLiteSupportStatus. The integer value
+  // corresponding to the TfLiteSupportStatus enum stored in the payload is
+  // extracted here to later map to the appropriate error code to be returned.
+  // In cases where the enum is not stored in (payload is NULL or the payload
+  // string cannot be converted to an integer), we set the error code value to
+  // be 1 (kError of TfLiteErrorCode used in the C library to signify any errors
+  // not falling into other categories.) Since payload is of type absl::Cord
+  // that can be type cast into an absl::optional<std::string>, we use the
+  // std::stoi function to convert it into an integer code if possible.
+  int generic_error_code = static_cast<int>(kError);
+  int error_code;
+  try {
+    // Try converting payload to integer if not payload is not empty. Otherwise
+    // convert a string signifying generic error code kError to integer.
+    error_code = std::stoi(static_cast<absl::optional<std::string>>(
+                               status.GetPayload(kTfLiteSupportPayload))
+                               .value_or(std::to_string(generic_error_code)));
```

Comment:
Can TfLiteSupportStatus be static_casted to an integer? The default underlying type of enum class is int.

Reply:
Sorry I am a little confused about the question.  The logic in these lines is as follows:
The payload stores the `TfLiteSupportStatus`. Payload is of type `absl::Cord`. The static casting in these lines is casting the [absl status payload](https://github.com/abseil/abseil-cpp/blob/master/absl/status/status.h#L566) to an optional `std::string`. This is fine. That's what I understood when I checked the [docs of absl::Cord](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/cord.h#L309). I guess, you can't directly cast payload to TfLiteSupportStatus since it's an int enum. So the string payload is passed to `stoi` if non null. If it is `null` a default string containing kError is passed to stoi. If direct conversion between payload and TfLiteSupportStatus was possible, we could have cast the payload integer directly to TfLiteSupportErrorCode for C.
Are you talking about static casting for enum values elsewhere?

Reply:
Ha,  I see.  I was misunderstood. Please ignore my previous comment.